### PR TITLE
Update operator-sdk to 1.1.0

### PR DESCRIPTION
**Pull request by miguelsorianod**
This PR updates operator-sdk to v1.1.0.

Pending:
- [x] Update licenses.xml
- [x] Decide and finish final state of metrics

### CODE REVIEW

**Comment by eguzki** [comment]
Looking good. 

Change is too big to review all in details. Hidden issues will show up and fixed later. 

Testing CRD's feature not yet implemented. Maybe we should consider bringing required source code to our repo. I find valuable testing we should not get lost.

**Comment by miguelsorianod** [comment]
- Reintroduced CRD tests and reenabled them in CircleCI
- Create specific test-type Makefile target. Go package filtering is done in the Makefile task to achieve that. In case we start to mix different test types in the same package (has not happened yet) a new approach will have to be followed.
- Replaced `verify-manifest` Makefile target with `validate-bundle`. Also done in CircleCI and reenabled job there.

**Comment by miguelsorianod** [comment]
- Licenses content updated and reenabled CircleCI job

**Comment by miguelsorianod** [comment]
Added permissions to manage APIcast finalizers:

There was an error where you cannot set an OwnerReference to an object if you don't have permissions to set finalizers on the Owner object type:

```
 "error": "secrets \"apicast-gateway-configuration\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on
```

**Comment by miguelsorianod** [comment]
Changed the `anyOf` OpenAPI validation section of the APIcast crd for the adminPortalCredentialsRef and embeddedConfigurationSecretRef fields.

Starting from newer k8s versions the OpenAPI validation has to follow a structural schema organization (see https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema)

Before this change when you tried to deploy the APIcast CRD the following message was received:
```
CustomResourceDefinition.apiextensions.k8s.io "apicasts.apps.3scale.net" is invalid: [
spec.validation.openAPIV3Schema.properties[spec].anyOf[0].properties[adminPortalCredentialsRef].type: Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].anyOf[1].properties[embeddedConfigurationSecretRef].type: Forbidden: must be empty to be structural
```

To fix this, the `anyOf` validation has been changed from:
```
          anyOf:
           - properties:
               adminPortalCredentialsRef:
                 type: object
             required: ["adminPortalCredentialsRef"]
           - properties:
               embeddedConfigurationSecretRef:
                 type: object
             required: ["embeddedConfigurationSecretRef"]
```

to:
```
          anyOf:
           - required: ["adminPortalCredentialsRef"]
           - required: ["embeddedConfigurationSecretRef"]
```

To follow the structural schema specification

File: doc/development.md [hunk 1]
```diff
@@ -76,167 +79,124 @@ active context
...
-make e2e-local-run
+make test-unit
 ```
 
-## Manifest management
+#### Run end-to-end tests
 
-`operator-courier` is used for metadata syntax checking and validation.
-This can be installed directly from pip:
+Access to a Kubernetes v1.11.3+ cluster required.
```
Comment (eguzki) [inline doc/development.md:172]: Check requirement

**Review by eguzki — CHANGES_REQUESTED** [review]

**Comment by miguelsorianod** [comment]
Fixed CRD test pointing to the correct bundle/manifests directory to get the CRD to validate

File: doc/development.md (hunk 1 above)
Comment (miguelsorianod) [reply doc/development.md:172]: It is not strictly needed. I'll remove it :+1:

Comment (miguelsorianod) [reply doc/development.md]: Fixed.

**Comment by miguelsorianod** [comment]
Ready for review again @eguzki

**Review by eguzki — APPROVED** [review]
Let the hunger games start!

**Pull request merged**